config/v3_5_experimental/types/config.go

```diff
@@ -61,5 +65,82 @@ func (cfg Config) Validate(c path.ContextPath) (r report.Report) {
 			r.AddOnError(c.Append("storage", "links", i, "path"), errors.ErrPathConflictsSystemd)
 		}
 	}
-	return
+
+	r.Merge(cfg.validateParents(c))
+
+	return r
+}
+
+func (cfg Config) validateParents(c path.ContextPath) report.Report {
```

Comment:
This seems to be the path that makes the Windows test fail:

```
github.com/coreos/ignition/v2/config/v3_5_experimental/types.Config.validateParents({{{{0x0, 0x0, 0x0}, {0x0, {...}, 0x0, {...}}}, {0x0, 0x0, {0x0, ...}}, ...}, ...}, ...)
	D:/a/ignition/ignition/config/v3_5_experimental/types/config.go:118 +0x1371
```

I wonder if the loops  in this function are taking more than 60s... one thing I would try to test is:

Maybe change the test to have more time?

I see CI calling:
```
Run go test -timeout 60s -cover \
```
So maybe trying 120 or 240 and seeing how much it actually takes. Then if that passes the tests, I would analyze the loops here to see if it can be re-written to be faster somehow.

I might be wrong, but that is my initial hunch looking at the failure and a pass on the code.

Reply:
That's weird because when I run it locally I am having a pass, look:

```
[ydesouza@ydesouza-thinkpadt14gen3 v3_5_experimental]$ go test -timeout 60s -cover 
PASS
        github.com/coreos/ignition/v2/config/v3_5_experimental  coverage: 91.3% of statements
ok      github.com/coreos/ignition/v2/config/v3_5_experimental  0.007s
```

(Anyways, I will try to improve the code's performance. )

Reply:
I tried with the path and I had this problem:

```
[ydesouza@ydesouza-thinkpadt14gen3 config]$ go test /home/ydesouza/workspace/rpms/ignition_project/github/fork/ignition/config/v3_5_experimental/types/config.go -timeou
t 60s -cover
# command-line-arguments
v3_5_experimental/types/config.go:38:11: undefined: Config
v3_5_experimental/types/config.go:74:11: undefined: Config
FAIL
```

Not sure why `Config` is "undefined" since it was already being used before I started the changes.

Reply:
I tried a few different approaches. Tried with the test function name, the function name itself and also with the main function in the config.go file to see it would have a different output.
```
[ydesouza@ydesouza-thinkpadt14gen3 ignition]$ go test ./config/v3_5_experimental -run validateParents
ok      github.com/coreos/ignition/v2/config/v3_5_experimental  (cached) [no tests to run]
```
```

[ydesouza@ydesouza-thinkpadt14gen3 ignition]$ go test ./config/v3_5_experimental -run Validate
ok      github.com/coreos/ignition/v2/config/v3_5_experimental  (cached) [no tests to run]
```

```
[ydesouza@ydesouza-thinkpadt14gen3 ignition]$ go test ./config/v3_5_experimental -run TestConfigValidation
ok      github.com/coreos/ignition/v2/config/v3_5_experimental  0.002s [no tests to run]
```

I also tried to run inside the v3_5_experimental directory and had this same output in each case:
```

[ydesouza@ydesouza-thinkpadt14gen3 v3_5_experimental]$ go test /home/ydesouza/workspace/rpms/ignition_project/github/fork/ignition/config/v3_5_experimental/types/config.go -run validateParents
# command-line-arguments
types/config.go:38:11: undefined: Config
types/config.go:74:11: undefined: Config
FAIL
```

I will read those docs and improve the code, thanks for sharing!

Reply:
I am not sure how to improve the loops, I was not able to iterate the way I wanted, making only a loop for files, directories and links. But I added some helper functions to improve performance.